controllers/release/release_controller.go

```diff
@@ -146,6 +151,12 @@ func setupControllerWithManager(manager ctrl.Manager, reconciler *Reconciler) er
 
 	return ctrl.NewControllerManagedBy(manager).
 		For(&v1alpha1.Release{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Watches(&source.Kind{Type: &applicationapiv1alpha1.SnapshotEnvironmentBinding{}}, &libhandler.EnqueueRequestForAnnotation{
+			Type: schema.GroupKind{
+				Kind:  "Release",
+				Group: "appstudio.redhat.com",
+			},
+		}, builder.WithPredicates(gitops.DeploymentStatusChangedPredicate(), goodies.GenerationUnchangedOnUpdatePredicate{})).
```

Comment:
Should the order be reversed? That way we filter more items I think.

Reply:
? I got this from you. I didn't know order mattered here. What does order do?

Reply:
My example was with a different trivial thing.

I haven't seen the code and I'm assuming predicates are executed in order so in this case the DeploymentStatusChangedPredicate would be executed first and would be applied for every single event. Then, the other predicate would be applied with the filtered events.

I have the impression that it would be more efficient to do it the other way around.

Reply:
Sure thing, will swap them